src/lib.rs

```diff
@@ -323,7 +331,14 @@ impl Distribution<SecretKey> for Standard {
     /// which uses [`rand::thread_rng()`](https://docs.rs/rand/0.6.1/rand/fn.thread_rng.html)
     /// internally as its RNG.
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> SecretKey {
-        SecretKey(Box::new(rng.gen04()))
+        // any way to use Fr::random(rng)?
```

Comment:
Why does
```rust
SecretKey(Box::new(Fr::random(rng)))
```
not work?
(I also wonder whether we can get rid of the `Box`, now that we're using `Zeroize`?)

Reply:
Since `?Size` trait bound in `sample`'s signature, you can only use a `Sized` rng instead

Reply:
Or we can add `sample` function ourselves, without implementing `Distribution`

Reply:
```rust
    pub fn sample<R: RngCore>(&self, rng: &mut R) -> Self {
        Self(Fr::random(rng))
    }
```